client/purplecaffeine/core.py

```diff
@@ -471,6 +471,9 @@ def list(
             with open(path, "r", encoding="utf-8") as trial_file:
                 trial_dict = json.load(trial_file, cls=TrialDecoder)
                 trials.append(Trial(**trial_dict))
+
+        if query:
```

Comment:
The problem here is you applied your query after the limit offset. We should be able to apply the limit to the list return by the query